### PR TITLE
feat: metadata exporter basic concept

### PR DESCRIPTION
**Pull request by assafchamoy**

### CODE REVIEW

File: src/common/i18n/en.json [hunk 1]
```diff
@@ -40,6 +40,7 @@
   "action.system-jobs.tooltip": "Display System Jobs",
   "action.system-core-info.tooltip": "Display system core info",
   "action.edit.tooltip": "Edit",
+  "action.export-metadata.tooltip": "Export Metadata",
```
Comment (alebinson) [inline src/common/i18n/en.json:43]: Terminology should be **Save**

File: src/common/i18n/he.json [hunk 2]
```diff
@@ -40,6 +40,7 @@
   "action.system-jobs.tooltip": "צפייה במשימות הוספה של המערכת",
   "action.system-core-info.tooltip": "צפייה במידע המערכת",
   "action.edit.tooltip": "עריכה",
+  "action.export-metadata.tooltip": "יצא קובץ מטא-דאטה",
```
Comment (alebinson) [inline src/common/i18n/he.json:43]: שמור ...

File: src/discrete-layer/components/layer-details/utils.ts [hunk 3]
```diff
@@ -243,6 +243,19 @@ export const extractUpdateRelatedFieldNames = (record: ILayerImage, descriptors:
   return updateRulesFields.map(field => field.fieldName) as string[];
 };
 
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function downloadJSONToClient(jsonObj: Record<any, any>, fileName: string): void {
+  const JSON_INDENTATION = 4;
+
+  const link = document.createElement('a');
```
Comment (alebinson) [inline src/discrete-layer/components/layer-details/utils.ts:251]: added <a> element should be deleted

**Review by alebinson — CHANGES_REQUESTED** [review]
see my comments

File: src/discrete-layer/models/userStore.ts [hunk 4]
```diff
@@ -66,6 +72,12 @@ const ROLES: IRole[] = [
       [UserAction.ENTITY_ACTION_LAYER3DRECORD_CREATE]: true,
       [UserAction.ENTITY_ACTION_LAYERDEMRECORD_CREATE]: false,
       [UserAction.ENTITY_ACTION_BESTRECORD_CREATE]: true,
+      [UserAction.ENTITY_ACTION_LAYERRASTERRECORD_SAVE_METADATA]: true,
+      [UserAction.ENTITY_ACTION_BESTRECORD_SAVE_METADATA]: true,
+      [UserAction.ENTITY_ACTION_LAYER3DRECORD_SAVE_METADATA]: true,
+      [UserAction.ENTITY_ACTION_LAYERDEMRECORD_SAVE_METADATA]: true,
+      [UserAction.ENTITY_ACTION_BESTRECORD_SAVE_METADATA]: true,
```
Comment (EllaMartirosyan) [inline src/discrete-layer/models/userStore.ts:79]: Duplicate line - 76 & 79

File: src/discrete-layer/models/userStore.ts [hunk 5]
```diff
@@ -94,6 +106,12 @@ const ROLES: IRole[] = [
       [UserAction.ENTITY_ACTION_LAYER3DRECORD_CREATE]: false,
       [UserAction.ENTITY_ACTION_LAYERDEMRECORD_CREATE]: false,
       [UserAction.ENTITY_ACTION_BESTRECORD_CREATE]: false,
+      [UserAction.ENTITY_ACTION_BESTRECORD_SAVE_METADATA]: true,
+      [UserAction.ENTITY_ACTION_LAYERRASTERRECORD_SAVE_METADATA]: true,
+      [UserAction.ENTITY_ACTION_LAYER3DRECORD_SAVE_METADATA]: true,
+      [UserAction.ENTITY_ACTION_LAYERDEMRECORD_SAVE_METADATA]: true,
+      [UserAction.ENTITY_ACTION_BESTRECORD_SAVE_METADATA]: true,
```
Comment (EllaMartirosyan) [inline src/discrete-layer/models/userStore.ts:113]: Duplicate line - 109 & 113

File: src/discrete-layer/views/components/action-resolver.component.tsx [hunk 6]
```diff
@@ -114,6 +114,24 @@ export const ActionResolver: React.FC<ActionResolverComponentProps> = observer((
...
+        case 'LayerDemRecord.save-metadata':
+          downloadJSONToClient(data, 'metadata.json');
+          break;
+        case 'BestRecord.save-metadata':
+          downloadJSONToClient(data, 'metadata.json');
+          break;
+        case 'VectorBestRecord.save-metadata':
+          downloadJSONToClient(data, 'metadata.json');
+          break;
+        case 'QuantizedMeshBestRecord.save-metadata':
```
Comment (EllaMartirosyan) [inline src/discrete-layer/views/components/action-resolver.component.tsx:132]: You can combine multiple cases with the same result

**Review by alebinson — APPROVED** [review]
gr8!

**Pull request merged**